src/main/java/programminglife/gui/controller/GraphController.java

```diff
@@ -385,19 +385,27 @@ public double getLocationCenterY() {
      *
      * @param nodeId is the node to center on.
      */
-    private void centerOnNodeId(int nodeId) {
+    private int centerOnNodeId(int nodeId) {
         DrawableNode drawableCenterNode = subGraph.getNodes().get(nodeId);
+        int centerId = nodeId;
 
         Bounds bounds = canvas.getParent().getLayoutBounds();
         double boundsHeight = bounds.getHeight();
         double boundsWidth = bounds.getWidth();
 
-        double xCoordinate = drawableCenterNode.getCenter().getX();
+        double xCoordinate;
+        if (drawableCenterNode != null) {
+            xCoordinate = drawableCenterNode.getCenter().getX();
+        } else {
+            centerId = graph.getChildIDs(nodeId)[0];
+            xCoordinate = subGraph.getNodes().get(centerId).getCenter().getX();
```

Comment:
wil subGraph.getNodes.iterator.next() return the next non SNPable node in the subgraph?